javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/internal/ExperimentalInstrumentationModule.java

```diff
@@ -70,9 +71,9 @@ default List<String> agentPackagesToHide() {
    * to get a reference to the module. <br>
    * Map value is a list of packages to open in the module
    *
-   * @return map of "witness class" FQN as key, list of packages as value.
+   * @return map of module to open as key, list of packages as value.
    */
-  default Map<String, List<String>> jpmsModulesToOpen() {
+  default Map<JavaModule, List<String>> jpmsModulesToOpen() {
```

Comment:
The module to open (and the classes it contains) are not necessarily visible from the classloader of the `InstrumentationModule` and IINM it is not possible to construct `JavaModule` without the loaded class at hand. Therefore unfortunately we have to use the classname as string here I think?

Reply:
Alternatively, we could provide our own `Module` class which is a lazy-`JavaModule` and can be constructed either using a string or a loaded class

Reply:
This construct is used in only once place and in this particular case the class is visible.

Reply:
For the current usage yes - because we only use this for `java.*` at the moment which is accessible for everywhere. But this should not be a requirement for the instrumentation API, which using `JavaModule` effectively results in.

E.g. what if you instrument some custom library which is not on the bootstrap and uses Java Modules for encapsulation?

Reply:
I think that at this point we don't need to focus on use cases that may or may not materialize in the future.

Reply:
For now this method is experimental so there is no issue to keep it as-is for now, however when we'll make it public that could be an issue, I'll add a TODO to the method so we don't forget about it.